core/src/main/java/org/infinispan/configuration/cache/IndexWriterConfiguration.java

```diff
@@ -14,9 +14,9 @@ public class IndexWriterConfiguration extends ConfigurationElement<IndexWriterCo
    public static final AttributeDefinition<Integer> INDEX_THREAD_POOL_SIZE =
          AttributeDefinition.builder(Attribute.THREAD_POOL_SIZE, 1, Integer.class).immutable().build();
    public static final AttributeDefinition<Integer> INDEX_QUEUE_COUNT =
-         AttributeDefinition.builder(Attribute.QUEUE_COUNT, 1, Integer.class).immutable().build();
+         AttributeDefinition.builder(Attribute.QUEUE_COUNT, 16, Integer.class).immutable().build();
```

Comment:
Note:

> raising this number above the [number of threads](https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#backend-lucene-threads) is never useful, as the number of threads limits how many queues can be processed in parallel.

Source: https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#backend-lucene-indexing-queues

So you may want to have this capped by the size of the thread pool. Now what is the size of your thread pool... who knows. Not me :)

Reply:
We don't use the thread pool provided by Hibernate Search, so I don't think (but of course I may be wrong)  we have some effect changing this value. Actually I think we should also remove this configuration property.

Reply:
The blocking queue default size is 150, but is obviously configurable by the user. https://github.com/infinispan/infinispan/blob/main/core/src/main/java/org/infinispan/factories/KnownComponentNames.java#L44

Reply:
It's not about the size of your threadpool's queue, but really the number of threads in your thread pool, because processing queues is completely blocking (and there's nothing I can do about it, it's Lucene).

If you have only 2 threads available in your pool, there's no use splitting indexing work into 16 different queues: you'll only ever process 2 queues at the same time. So you'd better use 2 queues at most.

Reply:
Whoops yes, I mistyped. The thread pool size is 150 and the queue is 5_000.